odbc/src/api/data.rs

```diff
@@ -775,12 +766,27 @@ pub unsafe fn format_bson_data(
         }
         _ => {}
     }
+
+    let uuid_repr = match &mongo_handle.as_connection() {
```

Comment:
Please before you merge, can you double check this? 
I am very surprised and I did not expect a handle to be "castable" to different types.
It should either be a connection handle or a statement handle, etc... (a pointer to a struct).
https://learn.microsoft.com/en-us/sql/odbc/reference/develop-app/handles?view=sql-server-ver16

Is there a test which validates this above? I did not see one. It would have to be an integration test with a UUID representation not being the default.

Reply:
I'll write a few tests showing it. We do cast the handle quite a bit [here](https://github.com/mongodb/mongo-odbc-driver/blob/master/odbc/src/api/functions.rs#L1548-L1615).